core/src/main/java/org/apache/accumulo/core/iterators/OrIterator.java

```diff
@@ -80,59 +128,103 @@ public int compareTo(TermSource o) {
       // sorted after they have been determined to be valid.
       return this.iter.getTopKey().compareColumnQualifier(o.iter.getTopKey().getColumnQualifier());
     }
+
+    /**
+     * Converts the given {@code Range} into the correct {@code Range} for this TermSource (per this expected table structure) and then seeks this TermSource's
+     * SKVI.
+     */
+    public void seek(Range originalRange) throws IOException {
+      // the infinite start key is equivalent to a null startKey on the Range.
+      if (!originalRange.isInfiniteStartKey()) {
+        Key originalStartKey = originalRange.getStartKey();
+        // Pivot the provided range into the range for this term
+        Key newKey = new Key(originalStartKey.getRow(), term, originalStartKey.getColumnQualifier(), originalStartKey.getTimestamp());
+        // Construct the new range, preserving the other attributes on the provided range.
+        currentRange = new Range(newKey, originalRange.isStartKeyInclusive(), originalRange.getEndKey(), originalRange.isEndKeyInclusive());
```

Comment:
Are you sure you want to trust the start key  inclusivity flag if you are setting term in the new key above?  Seems like if you're adjusting the range you want to make your key inclusive. Am I wrong in my logic?